File: v-conf-c#/vehicle_config_c#/project_vc#/project_vc#/DTOs/ComponentDropdownDTO.cs
using System.Collections.Generic;

namespace project_vc_.DTOs;

public class ComponentDropdownDTO
{
    public int? BaseCompId { get; set; }
    public string? ComponentName { get; set; }
    public List<OptionDTO>? Options { get; set; }

    public ComponentDropdownDTO() { }
    public ComponentDropdownDTO(int? baseCompId, string? componentName, List<OptionDTO>? options)
    {
        BaseCompId = baseCompId;
        ComponentName = componentName;
        Options = options;
    }
}

File: v-conf-java/v-conf/v-conf/src/main/java/com/example/dto/ComponentDropdownDto.java
package com.example.dto;

import java.util.List;

public class ComponentDropdownDto {

    private Integer baseCompId;      // ✅ REQUIRED
    private String componentName;
    private List<OptionDto> options;

    // ✅ New constructor (recommended)
    public ComponentDropdownDto(Integer baseCompId, String componentName, List<OptionDto> options) {
        this.baseCompId = baseCompId;
        this.componentName = componentName;
        this.options = options;
    }

    // ✅ Optional: keep no-arg constructor if Jackson needs it
    public ComponentDropdownDto() {
    }

    public Integer getBaseCompId() {
        return baseCompId;
    }

    public void setBaseCompId(Integer baseCompId) {
        this.baseCompId = baseCompId;
    }

    public String getComponentName() {
        return componentName;
    }

    public void setComponentName(String componentName) {
        this.componentName = componentName;
    }

    public List<OptionDto> getOptions() {
        return options;
    }

    public void setOptions(List<OptionDto> options) {
        this.options = options;
    }
}
